extension/src/popup/components/ModalInfo/index.tsx

```diff
@@ -1,42 +1,77 @@
 import React from "react";
 import classNames from "classnames";
 import { Card, Icon } from "@stellar/design-system";
+import { useTranslation } from "react-i18next";
 
 import { PunycodedDomain } from "popup/components/PunycodedDomain";
-import { MaliciousDomainWarning } from "../WarningMessages";
+import IconShieldPlus from "popup/assets/icon-shield-plus.svg";
 
 import "./styles.scss";
 
+export type PillType = "Connection" | "Trustline" | "Transaction";
+
+interface PillyCopyProps {
+  pillType: PillType;
+}
+
+const PillCopy = ({ pillType }: PillyCopyProps) => {
+  const { t } = useTranslation();
+
+  if (pillType === "Transaction") {
+    return (
+      <>
+        <Icon.Link />
+        <div>{t("Transaction Request")}</div>
+      </>
+    );
+  }
+
+  if (pillType === "Trustline") {
+    return (
+      <>
+        <img src={IconShieldPlus} alt="Add trustline icon" />
+        <div>{t("Add Asset trustlinet")}</div>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <Icon.Link />
+      <div>{t("Connection Request")}</div>
+    </>
```

Comment:
I think this will end up colliding a bit with the domain check in this screen: https://www.figma.com/design/C3G0a4Gd6RQyplRBppGDsL/Freighter-(SDS-v3)?node-id=2051-33657&node-type=FRAME&m=dev

But we can iron that out